### ISSUE TITLE
team_ids are detected as changed after project creation

### ISSUE THREAD

**Issue by phillipuniverse** [rollbar/terraform-provider-rollbar#251]
Terraform version - 1.0.11
Rollbar provider version - 1.4.0

1. Create a project using Terraform
2. Re-run the plan with `terraform plan -no-color -out plan`

```
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # rollbar_project.pando has been changed
  ~ resource "rollbar_project" "pando" {
      ~ date_modified = 1637152947 -> 1637154343
        id            = "528882"
        name          = "pando"
      ~ team_ids      = [
          - 376208,
        ]
        # (3 unchanged attributes hidden)
    }

Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes, the following plan may include actions to undo or respond to these changes.

──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

No changes. Your infrastructure matches the configuration.

Your configuration already matches the changes detected above. If you'd like to update the Terraform state to match, create and apply a refresh-only plan:
  terraform apply -refresh-only
```

The `date_modified` is also being tracked here, but the removal of the team ids is very peculiar. If I look at the project settings I can see the team is still assigned (the 'Everyone' team corresponds to 376208)

![image](https://user-images.githubusercontent.com/684275/142224364-38a8685b-e638-4ca6-be82-00c06ec3eccf.png)

**Comment by phillipuniverse** [comment]
The `date_modified` change might correspond to https://github.com/hashicorp/terraform/issues/28803

**Comment by jtsaito** [comment]
We are having the same issue.

I believe it is caused not by the Terraform provider or API client but. Instead, in our case it seems to be caused by the [API endpoint for listing a team's projects](https://explorer.docs.rollbar.com/#operation/list-a-teams-projects) returning incomplete data.

When a new project is created or an existing one is updated, API calls are made to set the team's projects ([AssignTeamToProject](https://github.com/rollbar/terraform-provider-rollbar/blob/6a3098b4391a0b3946a6cfab6121ff19154fc852/rollbar/resource_project.go#L145) and [UpdateProjectTeams](https://github.com/rollbar/terraform-provider-rollbar/blob/6a3098b4391a0b3946a6cfab6121ff19154fc852/rollbar/resource_project.go#L206)).

The operations (update/create) then call [resourceProjectRead](https://github.com/rollbar/terraform-provider-rollbar/blob/6a3098b4391a0b3946a6cfab6121ff19154fc852/rollbar/resource_project.go#L156) which is best practice for Terraform providers.

From there the stack trace is as follows:

- [call FindProjectTeamIDs](https://github.com/rollbar/terraform-provider-rollbar/blob/6a3098b4391a0b3946a6cfab6121ff19154fc852/rollbar/resource_project.go#L184)
- [call ListTeamProjectIDs](https://github.com/rollbar/terraform-provider-rollbar/blob/6a3098b4391a0b3946a6cfab6121ff19154fc852/client/project.go#L222) 
- [call pathTeamProjects](https://github.com/rollbar/terraform-provider-rollbar/blob/6a3098b4391a0b3946a6cfab6121ff19154fc852/client/team.go#L285)
- is [defined as](https://github.com/rollbar/terraform-provider-rollbar/blob/db1f5862ed0bceddb14241259fad15dcd0c83dbb/client/paths.go#L39) `/api/1/team/{teamID}/projects`

The API endpoint called is [documented here](https://explorer.docs.rollbar.com/#tag/Teams-Projects).

I ran a query against this endpoint to list the Rollbar Teams for our Rollbar organisation for a  particular project.

This returned a list of 20  team projects.

This is obviously wrong, because the web console shows 127 projects for the team (which is the number of projects actually assigned).

So, in our case the team-projects endpoint used in the `rollbar_projects` `create` and `update` implementation returns a ("very") incomplete list of projects. This results in the project's team ids not being set producing a non-empty diff.

I checked the [documentation on Rollbar API (1.0)](https://explorer.docs.rollbar.com/) for missing pagination, but I could not find any documentation on pagination for this endpoint or any other.

**Comment by jtsaito** [comment]
Rollbar support kindly told me the endpoint does support pagination. I checked and the Rollbar client does not, but subsequently I found there was a PR addressing the issue: https://github.com/rollbar/terraform-provider-rollbar/pull/229.

**Comment by cjolif** [comment]
The PR was finally merged in 1.5 *but* it does not seem to be solving the issue on our side.
